src/commons/sideContent/SideContentContestVoting.tsx

```diff
@@ -17,79 +17,183 @@ type StateProps = {
     isValid,
     handleContestEntryClick,
-    handleVotingSubmissionChange,
-    maxScore,
-    minScore
+    handleVotingSubmissionChange
   } = props;
   const [showContestEntries, setShowContestEntries] = useState<boolean>(true);
+  const [currentDraggedItem, setCurrentDraggedItem] = useState<HTMLElement | null>(null);
+  const [hoveredTier, setHoveredTier] = useState<string | null>(null);
+
+  const sortedContestEntries = useMemo(() => {
+    return [...contestEntries].sort((a, b) => a.submission_id - b.submission_id);
+  }, [contestEntries]);
+
+  const handleDragStart: React.DragEventHandler = e => {
+    setCurrentDraggedItem(e.currentTarget as HTMLElement);
+    e.dataTransfer.effectAllowed = 'move';
+  };
+
+  const handleDragEnd = useCallback(
+    (contestEntry: ContestEntry): React.DragEventHandler =>
+      e => {
+        const tierElement = (e.currentTarget as HTMLElement).closest('.tier');
+        if (tierElement) {
+          const tierName = tierElement.id.split('-')[1];
+          const tier = TIERS.find(t => t.name.toLowerCase() === tierName);
+          handleVotingSubmissionChange(contestEntry.submission_id, tier?.score || 0);
+        }
+        setHoveredTier(null);
+      },
+    [handleVotingSubmissionChange]
+  );
+
+  const handleDragOver = useCallback((e: React.DragEvent) => {
+    e.preventDefault();
+    e.dataTransfer.dropEffect = 'move';
+  }, []);
+
+  const handleDragEnter = useCallback((e: React.DragEvent): void => {
+    // added setTimeout here to give handleDragLeave time to execute if a dragenter event is about to be fired
+    setTimeout(() => {
+      const tierElement = (e.target as HTMLElement).closest('.tier');
+      if (tierElement) {
+        setHoveredTier(tierElement.id);
+      }
+    }, 0);
+  }, []);
+
+  const handleDragLeave = useCallback(
+    (e: React.DragEvent): void => {
+      const tierElement = (e.target as HTMLElement).closest('.tier');
+      if (tierElement && tierElement.id === hoveredTier) {
+        setHoveredTier(null);
+      }
+    },
+    [hoveredTier]
+  );
+
+  const handleDrop = useCallback(
+    (e: React.DragEvent): void => {
+      e.preventDefault();
+      const container = (e.target as HTMLElement).closest('.item-container');
+      if (container) {
+        container.appendChild(currentDraggedItem as Node);
+      }
+    },
+    [currentDraggedItem]
+  );
+
+  const tierBoard = useMemo(() => {
+    return TIERS.map(tier => (
+      <div
+        className={classNames('tier', {
+          'hovered-tier': hoveredTier === `tier-${tier.name.toLowerCase()}`
+        })}
+        key={`tier-${tier.name.toLowerCase()}`}
+        id={`tier-${tier.name.toLowerCase()}`}
+      >
+        <h2 style={{ backgroundColor: tier.color }}>{tier.name}</h2>
+        <div
+          className="item-container"
+          onDragOver={handleDragOver}
+          onDragLeave={handleDragLeave}
+          onDragEnter={handleDragEnter}
+          onDrop={handleDrop}
+        ></div>
```

Comment:
Since there's no children, let's clean this up a little by making the div self-closing.

```suggestion
        />
```

Reply:
Resolved!